crates/agglayer-certificate-orchestrator/src/lib.rs

```diff
@@ -405,17 +408,104 @@ where
+                // determine the following:
+                //
+                // - 1. If the state doesn't know the network and the height is 0, we update the
+                //   state. This is the first certificate for this network.
+                // - 2. If the state knows the network and the height is the next one, we update
+                //   the state. This is the next certificate for this network.
+                //
+                // - 3. If the state doesn't know the network and the height is not 0, we ignore
+                //   the proof.
+                // - 4. If the state knows the network and the height is not the next expected
+                //   one, we ignore the proof.
+                //
+                // When a generated proof is accepted:
+                // - We update the cursor for the network.
+                // - We update the latest proven certificate for the network.
+                // - We do not remove the pending certificate. (as it needs to be included in an
+                // epoch)
+                // - We spawn the next certificate for the network.
+
                 let entry = self.cursors.entry(network);
 
                 match entry {
+                    // - 1. If the state doesn't know the network and the height is 0, we update the
+                    //   state. This is the first certificate for this network.
                     Entry::Vacant(entry) if height == 0 => {
                         entry.insert(0);
+                        // TODO: Handle error if fails to set the latest proven certificate
+                        if let Err(error) = self
+                            .pending_store
+                            .set_latest_proven_certificate_per_network(
+                                &network,
+                                &height,
+                                &certificate.hash(),
+                            )
+                        {
+                            error!(
+                                "Failed to set the latest proven certificate per network: {:?}",
+                                error
+                            );
+                        }
+
+                        if let Err(error) = self.state_store.update_certificate_header_status(
+                            &certificate.hash(),
+                            &CertificateStatus::Proven,
+                        ) {
+                            error!(
+                                "Failed to update the certificate header status: {:?}",
+                                error
+                            );
+                        }
+                    }
+
+                    // - 2. If the state knows the network and the height is the next one, we update
+                    //   the sate. This is the next certificate for this network.
+                    Entry::Occupied(mut entry) if *entry.get() + 1 == height => {
+                        entry.insert(height);
```

Comment:
Looks like there is some code duplication between the two branches:
* `Entry::Vacant(entry) if height == 0`
* `Entry::Occupied(mut entry) if *entry.get() + 1 == height`

Reply:
Yes, it is going to be refactored in another PR.